lib/course_planner/tasks.ex

```diff
@@ -2,17 +2,17 @@ defmodule CoursePlanner.Tasks do
   @moduledoc false
   alias CoursePlanner.Repo
   alias CoursePlanner.Tasks.Task
-  alias Ecto.{DateTime, Changeset}
+  alias Ecto.Changeset
   import Ecto.Query, except: [update: 2]
 
-  @tasks from t in Task, where: is_nil(t.deleted_at), preload: [:user]
-
   def all do
-    Repo.all(@tasks)
+    Task
+    |> Repo.all()
+    |> Repo.preload(:user)
```

Comment:
I don't think you should preload a resource in a `all` call.
Make a separate query function to preload, put a parameter with preloading list or preload outside of the query.